modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -168,7 +169,11 @@ object CrossSources {
       (preprocessedInputFromArgs ++ preprocessedSourcesFromDirectives).distinct
 
     val preprocessedWithUsingDirs = preprocessedSources.filter(_.directivesPositions.isDefined)
-    if (preprocessedWithUsingDirs.length > 1) {
+    if (
+      preprocessedWithUsingDirs.length > 1 && !suppressDirectivesInMultipleFilesWarning.getOrElse(
```

Comment:
As per the previous comment, the whole `if` clause producing the warning could potentially be moved after the `buildOptions` are initialised.
and then you should be able to:
```suggestion
      preprocessedWithUsingDirs.length > 1 && !buildOptions.suppressWarningOptions.suppressDirectivesInMultipleFilesWarning.getOrElse(
```

Reply:
I think this won't work here this is what I found out the varaible `buildOptions` looks like during the warning suppression test:
<img width="1376" alt="image" src="https://user-images.githubusercontent.com/61919032/212093410-ed22eb50-955e-48a5-a942-01f064bbb78d.png">

Looks like it has no values for our suppress warning flag

Reply:
right, as those `buildOptions` are coming from processed sources... so they'd only include things from preprocessors, like `using` directives... sorry, I got confused there.
Nevermind my suggestion then.